test-d/iterable-element.ts

```diff
@@ -0,0 +1,20 @@
+import {IterableElement} from '../source/iterable-element';
+import {expectType} from 'tsd';
+
+declare const iterableElement: IterableElement<ReturnType<typeof secretGenerator>>;
+expectType<1 | 'two'>(iterableElement);
+
+declare const iterableElementAsync: IterableElement<ReturnType<typeof secretGeneratorAsync>>;
+expectType<true | Date>(iterableElementAsync);
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```

Comment:
Why are you disabling this rule?

Reply:
Well, xo fails when there is no return type specified. I think not specifying a return type emphasizes the usefulness of the `IterableElement` type!